packages/ckeditor5-ui/src/panel/balloon/balloonpanelview.ts

```diff
@@ -387,6 +395,27 @@ export default class BalloonPanelView extends View {
 		this.listenTo( global.window, 'resize', () => {
 			this.attachTo( options );
 		} );
+
+		// Hide the panel if the target element is no longer visible.
+		if ( targetElement && !this._resizeObserver ) {
+			const checkVisibility = () => {
+				// If the target element is no longer visible, hide the panel.
+				if ( !targetElement.checkVisibility() ) {
+					this.hide();
```

Comment:
I don't know if this is the right method. Doesn't the `unpin()` fit better? I don't know how those methods are used from outside this class.

Reply:
I'll check that.

Reply:
I changed it, basically it works exactly the same as `hide` but it seems to more convenient to use `unpin` method in `_startPinning`.